src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -175,7 +175,14 @@ public void rename() {
                     Localization.lang("Cancel"));
 
             if (confirm) {
-                pdfCleanup.cleanup(entry);
+                Path targetFilePath = file.get().getParent().resolve(targetFileName);
```

Comment:
For me, this is too much logic in the UI. We really aim to have the logic separated from the GUI. See https://github.com/JabRef/jabref/wiki/High-Level-Documentation

It costed us much headaches to refactor JabRef and also costs us headaches now. However, this really increases testability and enables extraction of libraries.

Therefore, may I ask whether it is possible to move this logic into RenamePdfCleanup somehow? Meaning: Is it possible to get a getter for `targetFileAlreadyExsists()`?

In RenamePdfCleanup, we also have `pathsDifferOnlyByCase`, which is not respected by this code: https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/logic/cleanup/RenamePdfCleanup.java#L101